src/openrct2/object/RideObject.cpp

```diff
@@ -1146,3 +1078,88 @@ ShopItem RideObject::ParseShopItem(const std::string& s)
     auto result = ShopItemLookupTable.find(s);
     return (result != ShopItemLookupTable.end()) ? result->second : ShopItem::None;
 }
+
+// Converts legacy sprite groups into OpenRCT2 sprite groups
+void RideObject::ReadLegacySpriteGroups(rct_ride_entry_vehicle* vehicle, uint16_t spriteGroups)
+{
+    auto baseSpritePrecision = SpritePrecision::Sprites32;
+    if (vehicle->flags & VEHICLE_ENTRY_FLAG_USE_16_ROTATION_FRAMES)
+        baseSpritePrecision = SpritePrecision::Sprites16;
+    if (vehicle->flags & VEHICLE_SPRITE_FLAG_USE_4_ROTATION_FRAMES)
+        baseSpritePrecision = SpritePrecision::Sprites4;
+
+    if (spriteGroups & VEHICLE_SPRITE_FLAG_FLAT)
+    {
+        vehicle->SpriteGroups[static_cast<uint8_t>(SpriteGroupType::SlopeFlat)].spritePrecision = baseSpritePrecision;
+    }
+    if (spriteGroups & VEHICLE_SPRITE_FLAG_GENTLE_SLOPES)
+    {
+        vehicle->SpriteGroups[static_cast<uint8_t>(SpriteGroupType::Slopes12)].spritePrecision = SpritePrecision::Sprites4;
+        vehicle->SpriteGroups[static_cast<uint8_t>(SpriteGroupType::Slopes25)].spritePrecision = baseSpritePrecision;
```

Comment:
```suggestion
        vehicle->SpriteGroups[EnumValue(SpriteGroupType::Slopes12)].spritePrecision = SpritePrecision::Sprites4;
        vehicle->SpriteGroups[EnumValue(SpriteGroupType::Slopes25)].spritePrecision = baseSpritePrecision;
```

Also for all cases below.